### ISSUE TITLE
runtime: TestVDSO failures due to hang or crash in SignalInVDSO since 2022-09-02

### ISSUE THREAD

**Issue by bcmills** [golang/go#55073]
`greplogs -l -e 'FAIL: TestVDSO.*(?:\n\s+.*)*testprog\.exe SignalInVDSO' --since=2022-01-01`
[2022-09-06T15:49:30-c35bc69/darwin-arm64-11](https://build.golang.org/log/ff343c4817a9a3e40fc8d3a3a8fbd02dce20a452)
```
--- FAIL: TestVDSO (60.02s)
    testenv.go:468: [/tmp/buildlet/tmp/go-build1960920154/testprog.exe SignalInVDSO] exit status: exit status 2
    crash_test.go:144: output:
        SIGQUIT: quit
        PC=0x182fa8304 m=1 sigcode=0
        
…        
        goroutine 1 [runnable]:
        time.Time.Sub({0xc0bdf955c4d64fd0, 0x17f57176, 0x1004d19c0}, {0xc0bdf955a87e4d70, 0x29a82, 0x1004d19c0})
        	/tmp/buildlet/go/src/time/time.go:863 +0x18c fp=0x1400006ee40 sp=0x1400006ee40 pc=0x10039bb5c
        main.signalInVDSO()
        	/private/tmp/buildlet/go/src/runtime/testdata/testprog/vdso.go:36 +0x114 fp=0x1400006eed0 sp=0x1400006ee40 pc=0x1003e4654
        main.main()
        	/private/tmp/buildlet/go/src/runtime/testdata/testprog/main.go:34 +0x120 fp=0x1400006ef70 sp=0x1400006eed0 pc=0x1003e0dd0
        runtime.main()
        	/tmp/buildlet/go/src/runtime/proc.go:250 +0x248 fp=0x1400006efd0 sp=0x1400006ef70 pc=0x10035d9e8
        runtime.goexit()
        	/tmp/buildlet/go/src/runtime/asm_arm64.s:1172 +0x4 fp=0x1400006efd0 sp=0x1400006efd0 pc=0x10038ab94
```
[2022-09-06T15:48:16-07b19bf/linux-386-longtest](https://build.golang.org/log/27352b9797cee026330ed91d89714cd30abd0268)
```
##### maymorestack=mayMoreStackPreempt
--- FAIL: TestVDSO (0.15s)
    testenv.go:468: [/workdir/tmp/go-build2315874385/testprog.exe SignalInVDSO] exit status: signal: segmentation fault (core dumped)
    crash_test.go:144: output:
        
        
        wanted:
        success
FAIL
FAIL	runtime	95.444s
```
[2022-09-06T15:44:28-32f68b5/linux-386-longtest](https://build.golang.org/log/d90e0c539de6eebb75f671d427d5b6bcb705a9d2)
[2022-09-05T21:39:28-4c1ca42/linux-386-longtest](https://build.golang.org/log/223673cfa24268ed01b8c4f467f3c5496953a41c)
[2022-09-05T08:28:34-bd5595d/linux-386-longtest](https://build.golang.org/log/3920217184f02a19abd0bb80ad17696dee0684ce)
[2022-09-05T08:08:24-af7f417/linux-386-longtest](https://build.golang.org/log/7ef94db13f10645fca7c9e9113d76f0d381c824c)
[2022-09-05T08:08:18-4ad55cd/linux-386-longtest](https://build.golang.org/log/d8ef3808241bd7add1cbcc590066ca21829ccf6e)
[2022-09-05T08:07:47-357b922/linux-386-longtest](https://build.golang.org/log/ee0bf370cc054c33c7445df13a9f99b7fe580cd2)
[2022-09-05T07:17:56-4e7e7ae/linux-386-longtest](https://build.golang.org/log/c34d1e63bc1f8d3c36e51777813bcce1cc8ac95b)
[2022-09-05T07:14:08-3fbcf05/linux-386-longtest](https://build.golang.org/log/9d07ddf12ccc5c9880b691e63ffb78db85a90369)
[2022-09-04T04:17:04-535fe2b/linux-386-longtest](https://build.golang.org/log/0ee68885989d554fe8013c21870b9490b3682ca9)
[2022-09-03T18:21:45-a0f0582/linux-386-longtest](https://build.golang.org/log/8256dfbc548facaaefaba5433c9bdf8c04b6fc44)
[2022-09-03T15:45:36-f798dc6/linux-386-longtest](https://build.golang.org/log/51e88d345d457a4251ec8e38aabcc01e0e4ceee8)
[2022-09-02T19:22:26-0fda8b1/linux-386-longtest](https://build.golang.org/log/c16de76490d3c7b5121c72089010926f7e3f5ccb)
[2022-09-02T19:09:03-55ca6a2/linux-386-longtest](https://build.golang.org/log/8770366c112fd5c78cc44a3ec9b3da0438db34da)
[2022-09-02T19:08:56-b91e373/linux-386-longtest](https://build.golang.org/log/ae64ea2dab7a9a3941b3fa610c942328bca41b1f)
[2022-09-02T19:08:53-dbf442b/linux-386-longtest](https://build.golang.org/log/35640ead268e7600c4c2d45eff20bcf836a4eabd)

(attn @golang/runtime)

**Comment by bcmills** [comment]
This may be related to #54778.

**Comment by bcmills** [comment]
This may also have been fixed by #54885? (Not clear to me whether that issue explains the hang in the test on `darwin/arm64`.)
